tests/chainer_tests/functions_tests/connection_tests/test_deconvolution_2d.py

```diff
@@ -87,54 +110,67 @@ def setUp(self):
-        with chainer.using_config('cudnn_deterministic',
-                                  self.cudnn_deterministic):
-            y_cpu = F.deconvolution_2d(
-                x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+    def forward_cpu(self, inputs):
+        x, W, b = inputs
+        x_cpu = chainer.Variable(x)
+        W_cpu = chainer.Variable(W)
+        b_cpu = None if b is None else chainer.Variable(b)
+        y_cpu = F.deconvolution_2d(
+            x_cpu, W_cpu, b_cpu, stride=self.stride, pad=self.pad,
+            outsize=self.outsize)
+        return y_cpu,
+
+    def check_forward(self, inputs, backend_config):
+        y_expected, = self.forward_cpu(inputs)
+
+        if backend_config.use_cuda:
+            inputs = cuda.to_gpu(inputs)
+
+        x, W, b = inputs
+        x = chainer.Variable(x)
+        W = chainer.Variable(W)
+        b = None if b is None else chainer.Variable(b)
+
+        with backend_config:
+            y_actual = F.deconvolution_2d(
+                x, W, b, stride=self.stride, pad=self.pad,
                 outsize=self.outsize)
 
-        x_gpu = chainer.Variable(cuda.to_gpu(self.x))
-        W_gpu = chainer.Variable(cuda.to_gpu(self.W))
-        b_gpu = None if self.nobias else chainer.Variable(
-            cuda.to_gpu(self.b))
-        with chainer.using_config('use_cudnn', self.use_cudnn):
-            with chainer.using_config('cudnn_deterministic',
-                                      self.cudnn_deterministic):
-                with chainer.using_config('autotune', self.autotune):
-                    y_gpu = F.deconvolution_2d(
-                        x_gpu, W_gpu, b_gpu, stride=self.stride, pad=self.pad,
-                        outsize=self.outsize)
-
-        self.assertEqual(y_cpu.data.dtype, self.x_dtype)
-        self.assertEqual(y_gpu.data.dtype, self.x_dtype)
+        assert y_expected.data.dtype == self.x_dtype
+        assert y_actual.data.dtype == self.x_dtype
         testing.assert_allclose(
-            y_cpu.data, y_gpu.data.get(), **self.test_forward_options)
+            y_expected.data, y_actual.data.get(), **self.test_forward_options)
 
     @attr.gpu
-    def test_forward_consistency_im2col(self):
-        self.use_cudnn = 'never'
-        self.test_forward_consistency()
+    def test_forward(self, backend_config):
+        # Forward test does not currently target CPU backend.
+        # It only tests for consistency between GPU and CPU computation.
+        if not backend_config.use_cuda:
+            return
+        self.check_forward(self.inputs, backend_config)
+
+    def check_backward(self, inputs, grad_outputs, backend_config):
 
-    def check_backward(self, x_data, W_data, b_data, y_grad):
-        xp = cuda.get_array_module(x_data)
+        xp = backend_config.xp
+        if backend_config.use_cuda:
+            inputs = cuda.to_gpu(inputs)
+            grad_outputs = cuda.to_gpu(grad_outputs)
+
+        x_data, W_data, b_data = inputs
```

Comment:
This is a small detail but can we keep the variables names consistent with `check_forward()`, i.e. `x, W, b = inputs`? Oh, and I really like the idea of packing the inputs and gradients. A lot less clutter.

Reply:
Do I need to fix it here? Those code will be refactored in #3499 anyway.

Reply:
I see, thanks for the response.